@XD-DENG Just finishing up cleaning all the pylint/black highlighting stuff on my local. I am stuck on one last piece and wanted to ask for some advice/recommendation.

`airflow/providers/ssh/hooks/ssh.py:93:8: R1702: Too many nested blocks (4/3) (too-many-nested-blocks)` 

There's quite a few `if` nested blocks in that file and I feel like changing it up will kind of mess up the readability.

Thoughts?

--- a/airflow/providers/ssh/hooks/ssh.py
+++ b/airflow/providers/ssh/hooks/ssh.py
@@ -107,11 +107,12 @@ class SSHHook(BaseHook):
 
                 private_key = extra_options.get('private_key')
                 private_key_passphrase = extra_options.get('private_key_passphrase')
-                if private_key:
-                    if private_key_passphrase:
-                        self.pkey = paramiko.RSAKey.from_private_key(StringIO(private_key), password=private_key_passphrase)
-                    else:
-                        self.pkey = paramiko.RSAKey.from_private_key(StringIO(private_key))
+                if private_key and private_key_passphrase:
+                    self.pkey = paramiko.RSAKey.from_private_key(
+                        StringIO(private_key), password=private_key_passphrase
+                    )
+                elif private_key and not private_key_passphrase:
+                    self.pkey = paramiko.RSAKey.from_private_key(StringIO(private_key))
 
                 if "timeout" in extra_options:
                     self.timeout = int(extra_options["timeout"], 10)

--- a/tests/providers/ssh/hooks/test_ssh.py
+++ b/tests/providers/ssh/hooks/test_ssh.py
@@ -69,7 +69,7 @@ class TestSSHHook(unittest.TestCase):
         with create_session() as session:
             conns_to_reset = [
                 cls.CONN_SSH_WITH_PRIVATE_KEY_EXTRA,
-                cls.CONN_SSH_WITH_PRIVATE_KEY_PASSPHRASE_EXTRA
+                cls.CONN_SSH_WITH_PRIVATE_KEY_PASSPHRASE_EXTRA,
             ]
             connections = session.query(Connection).filter(Connection.conn_id.in_(conns_to_reset))
             connections.delete(synchronize_session=False)

--- a/tests/providers/ssh/hooks/test_ssh.py
+++ b/tests/providers/ssh/hooks/test_ssh.py
@@ -19,6 +19,9 @@ import json
 import unittest
 from io import StringIO
 
+from typing import Optional
+import random
+import string
 import mock
 import paramiko
 
@@ -26,10 +29,6 @@ from airflow.models import Connection
 from airflow.providers.ssh.hooks.ssh import SSHHook
 from airflow.utils import db
 from airflow.utils.session import create_session
-
-from typing import Optional
-import random
-import string
 
 HELLO_SERVER_CMD = """
 import socket, sys
@@ -57,6 +56,7 @@ TEST_PRIVATE_KEY = generate_key_string(pkey=TEST_PKEY)
 
 PASSPHRASE = ''.join(random.choice(string.ascii_letters) for i in range(10))
 TEST_ENCRYPTED_PRIVATE_KEY = generate_key_string(pkey=TEST_PKEY, passphrase=PASSPHRASE)
+
 
 class TestSSHHook(unittest.TestCase):
     CONN_SSH_WITH_PRIVATE_KEY_EXTRA = 'ssh_with_private_key_extra'
@@ -112,9 +112,7 @@ class TestSSHHook(unittest.TestCase):
                 host='localhost',
                 conn_type='ssh',
                 extra=json.dumps(
-                    {   "private_key": TEST_ENCRYPTED_PRIVATE_KEY,
-                        "private_key_passphrase": PASSPHRASE
-                    }
+                    {"private_key": TEST_ENCRYPTED_PRIVATE_KEY, "private_key_passphrase": PASSPHRASE}
                 ),
             )
         )
@@ -237,6 +235,7 @@ class TestSSHHook(unittest.TestCase):
                 host_pkey_directories=[],
                 logger=hook.log,
             )
+
     @mock.patch('airflow.providers.ssh.hooks.ssh.SSHTunnelForwarder')
     def test_tunnel_with_private_key_passphrase(self, ssh_mock):
         hook = SSHHook(

